tt_metal/impl/dispatch/kernels/command_queue_producer.cpp

```diff
@@ -64,7 +64,7 @@ void kernel_main() {
     constexpr uint32_t issue_queue_start_addr = get_compile_time_arg_val(1);
 
     // Only the issue queue size is a runtime argument
-    uint32_t issue_queue_size = get_arg_val<uint32_t>(0);
+    uint32_t issue_queue_size = get_compile_time_arg_val(2);
```

Comment:
this will need to be updated in the `remote_issue_queue_reader` as well. I think you may need to rebase your branch to grab those changes